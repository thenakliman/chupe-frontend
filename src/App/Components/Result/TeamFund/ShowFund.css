.team-fund-container {
  width: 65rem;
  margin: 0 auto;
}