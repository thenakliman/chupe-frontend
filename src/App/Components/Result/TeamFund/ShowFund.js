import PropTypes from 'prop-types';
import React from 'react';

/**
* Show Fund component of the application.
*
* @author: thenakliman
*/
export class ShowFund extends React.Component {
  /** Fetch team fund while mounting this component */
  componentWillMount() {
    this.props.fetchFunds(this.props.match.params.id);
  }

  /** Component for show funds.
   * @return {object} Return component for showing funds
   */
  render() {
    return (
      <div id='user-fund-container-id' className='team-fund-container'>
        <table id='fund-table-id'>
          <thead>
            <tr>
              <th>S.No</th>
              <th>Owner</th>
              <th>type</th>
              <th>Amount</th>
              <th>Added On</th>
            </tr>
          </thead>
          <tbody>
          { this.props.funds.map((fund, index) => (
              <tr key={`${fund.id}`}>
                <td>{index+1}</td>
                <td>{fund.owner}</td>
                <td>{fund.type}</td>
                <td>{'CREDIT' === fund.transactionType? fund.amount: -fund.amount}</td>
                <td>{fund.createdAt}</td>
              </tr>
            ))
          }
          </tbody>
        </table>
      </div>
    );
  }
}

ShowFund.propTypes = {
  fetchFunds: PropTypes.func.isRequired,
  funds: PropTypes.array,
};
