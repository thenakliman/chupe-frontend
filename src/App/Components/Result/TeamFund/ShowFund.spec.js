import React from 'react'; // eslint-disable-line no-unused-vars
import {ShowFund} from './ShowFund'; // eslint-disable-line no-unused-vars
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Show fund component snapshot', () => {
    it('should match the snapshot', () => {
        const match = {params: {id: 'test-owner'}};
        const funds = [
          {id: 10, amount: 10, transactionType: 'CREDIT'},
          {id: 11, amount: 10, transactionType: 'DEBIT'}
        ];

        const wrapper = shallow(<ShowFund fetchFunds={()=>{}}
                                          funds={funds}
                                          match={match}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('Show Fund component', () => {
    it('Should have container div for Team fund', () => {
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={()=>{}}
                                          funds={[]}
                                          match={match}/>);

        expect(wrapper.find('#user-fund-container-id').length).toEqual(1);
    });

    it('Should call fetch fund on mount', () => {
        const owner = 'test-owner';
        const match = {params: {id: owner}};
        const fetchFunds = jest.fn();
        shallow(<ShowFund fetchFunds={fetchFunds}
                          funds={[]}
                          match={match}/>);

        expect(fetchFunds).toHaveBeenCalledWith(owner);
    });

    it('Should have fund for each member', () => {
        const funds = [{id: 10}, {id: 11}];
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={() => {}}
                                          funds={funds}
                                          match={match}/>);

        expect(wrapper.find('tbody').children().length
            ).toEqual(2);
    });

    it('Should have table header for S.No', () => {
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={() => {}}
                                          funds={[]}
                                          match={match}/>);

        const hasUsername = wrapper.find('th').someWhere(
            (header) => header.props().children === 'S.No');
        expect(hasUsername).toBe(true);
    });

    it('Should have table header for Owner', () => {
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={() => {}}
                                          funds={[]}
                                          match={match}/>);

        const hasUsername = wrapper.find('th').someWhere(
            (header) => header.props().children === 'Owner');
        expect(hasUsername).toBe(true);
    });

    it('Should have table header for type ', () => {
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={() => {}}
                                          funds={[]}
                                          match={match}/>);

        const hasUsername = wrapper.find('th').someWhere(
            (header) => header.props().children === 'type');
        expect(hasUsername).toBe(true);
    });

    it('Should have table header for Amount', () => {
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={() => {}}
                                          funds={[]}
                                          match={match}/>);

        const hasUsername = wrapper.find('th').someWhere(
            (header) => header.props().children === 'Amount');
        expect(hasUsername).toBe(true);
    });

    it('Should have table header for Added On', () => {
        const match = {params: {id: 'test-owner'}};
        const wrapper = shallow(<ShowFund fetchFunds={() => {}}
                                          funds={[]}
                                          match={match}/>);

        const hasUsername = wrapper.find('th').someWhere(
            (header) => header.props().children === 'Added On');
        expect(hasUsername).toBe(true);
    });
});
