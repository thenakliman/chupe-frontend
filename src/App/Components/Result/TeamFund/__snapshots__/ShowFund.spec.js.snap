// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show fund component snapshot should match the snapshot 1`] = `
<div
  className="team-fund-container"
  id="user-fund-container-id"
>
  <table
    id="fund-table-id"
  >
    <thead>
      <tr>
        <th>
          S.No
        </th>
        <th>
          Owner
        </th>
        <th>
          type
        </th>
        <th>
          Amount
        </th>
        <th>
          Added On
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="10"
      >
        <td>
          1
        </td>
        <td />
        <td />
        <td>
          10
        </td>
        <td />
      </tr>
      <tr
        key="11"
      >
        <td>
          2
        </td>
        <td />
        <td />
        <td>
          -10
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
